Modules/_decimal/_decimal.c

```diff
@@ -3411,6 +3364,54 @@ dict_get_item_string(PyObject *dict, const char *key, PyObject **valueobj, const
     return 0;
 }
 
+/*
+ * Fallback _pydecimal formatting for new format specifiers that mpdecimal does
+ * not yet support. As documented, libmpdec follows the PEP-3101 format language:
+ * https://www.bytereef.org/mpdecimal/doc/libmpdec/assign-convert.html#to-string
+ */
+static PyObject *
+pydec_format(PyObject *dec, PyObject *context, PyObject *fmt, decimal_state *state)
+{
+    PyObject *result;
+    PyObject *pydec;
+    PyObject *rounding;
+    PyObject *caps;
+    PyObject *u;
+
+    if (state->PyDecimal == NULL) {
+        state->PyDecimal = _PyImport_GetModuleAttrString("_pydecimal", "Decimal");
+        if (state->PyDecimal == NULL) {
+            return NULL;
+        }
+    }
+
+    u = dec_str(dec);
+    if (u == NULL) {
+        return NULL;
+    }
+
+    pydec = PyObject_CallOneArg(state->PyDecimal, u);
+    Py_DECREF(u);
+    if (pydec == NULL) {
+        return NULL;
+    }
+
+    rounding = context_getround(context, NULL);
+    caps = context_getcapitals(context, NULL);
+    result = PyObject_CallMethod(pydec, "_fallback_format", "(OOO)", fmt, rounding, caps);
```

Comment:
`context_getcapitals()` can fail, so you need to check its result. It is easy to pass `CtxCaps(context)` to `PyObject_CallMethod()` and use format unit "i".

But if you have a context object, why not pass it directly to `__format__()`, without an intermediate method? From Python point of view, both C and Python implementations of Context should be compatible.